docs/doc_5_best_practices.md

```diff
@@ -28,6 +28,27 @@ if test != ""
     requestBody.StringPointerValue = test
 ```
 
+## Working with Date Fields
+
+In the Atlas SDK GO, date fields are represented by `*time.Time` type to handle time-related data. 
+When comparing values that are `time.Time` based never compare pointers directly or 
+
+1. Avoid using direct comparison operators (e.g., `myStruct.MyDateField == ""`) to check for equality. Comparing pointers directly will check if they refer to the same memory address rather than comparing the actual date values. Since each time.Time instance is allocated in different memory locations, direct comparisons may yield unexpected results.
+
+2. Use the `Has` Function for Checking Non-Nil Pointers:
+The SDK provides a dedicated `HasFieldName` or `GetFieldName` function for each model to check if a time.Time pointer is not nil before accessing its value. Always use this function to ensure that the pointer is valid before performing any operations.
```

Comment:
Kind of an unrelated question, but aren't the `hasFieldName` and `getFieldName` methods kind of pointless? Like, what's the difference between getting a `false` on `hasFieldName` vs getting `nil` when trying to access it to be sure of it's value? With `getFieldName` the difference is even smaller.

Reply:
Very good question.

> getFieldName pointless 

Golang requires boilerplate code for the unified handling of the pointer values. 
We strive for better UX by giving developers a unified way to access fields.
Example:
https://github.com/mongodb/atlas-sdk-go/blob/168798f1930c1fbc2484973f23a4fc904e2adec5/admin/model_dedicated_hardware_spec.go#L43-L49

We also provide direct structure access for advanced users. 

> difference between getting a false on hasFieldName vs getting nil

`if struct.value == nil` is not reflecting every case for nullability in golang. 
SDK take on nullability:
https://github.com/mongodb/atlas-sdk-go/blob/168798f1930c1fbc2484973f23a4fc904e2adec5/admin/utils.go#L39

We could improve it over time without affecting users.

Reply:
Since that implementation was never written by us (it came from the upstream) I think we can simplify it to some extent and maybe avoid reflection. Will create separate jira